makesyscalls.lua

```diff
@@ -37,18 +35,21 @@ local unistd = require("posix.unistd")
 
 local savesyscall = -1
 local maxsyscall = -1
+local structs = {}
 local generated_tag = "@" .. "generated"
 
 -- Default configuration; any of these may get replaced by a configuration file
 -- optionally specified.
 local config = {
-	os_id_keyword = "FreeBSD",
+	os_id_keyword = "FreeBSD",		-- obsolete, ignored on input, not generated
 	abi_func_prefix = "",
+	libsysmap = "/dev/null",
+	libsys_h = "/dev/null",
 	sysnames = "syscalls.c",
 	sysproto = "../sys/sysproto.h",
 	sysproto_h = "_SYS_SYSPROTO_H_",
 	syshdr = "../sys/syscall.h",
-	sysmk = "../sys/syscall.mk",
+	sysmk = "/dev/null",
```

Comment:
This is an example of something we can definitely generate on the fly.
The syscall.mk file just has one line that has a list of MIASM files. We can instead just have libsys generate it on the fly.

Reply:
I think the same is true of libsys_h, libsysmap, syscalls.c easily fall under this. sysproto.h may as well. I fear that we need to keep committing syscalls.h to the tree, though. And given how often sysproto.h is included, it likely needs to be committed, though we do fine with the vnode stuff getting things built early enough.